microsoft-azure-storage/src/com/microsoft/azure/storage/core/ExecutionEngine.java

```diff
@@ -146,27 +149,16 @@
                     result = task.postProcessResponse(request, parentObject, client, opContext, result);
                     Logger.info(opContext, LogConstants.POST_PROCESS_DONE);
 
-                    // Success return result and drain the input stream.
-                    if ((task.getResult().getStatusCode() >= 200) && (task.getResult().getStatusCode() < 300)) {
-                        if (request != null) {
-                            InputStream inStream = request.getInputStream();
-                            // At this point, we already have a result / exception to return to the user.
-                            // This is just an optimization to improve socket reuse.
-                            try {
-                                Utility.writeToOutputStream(inStream, null, -1, false, false, null,
-                                        task.getRequestOptions());
-                            }
-                            catch (final IOException ex) {
-                            }
-                            catch (StorageException e) {
-                            }
-                            finally {
-                                inStream.close();
-                            }
-                        }
+                    // At this point, we already have a result / exception to return to the user.
+                    // This is just an optimization to improve socket reuse.
+                    if (task.getResult().getStatusCode() < HttpURLConnection.HTTP_BAD_REQUEST) {
```

Comment:
@jofriedm-msft - this new condition matches Java's internal logic, inside `sun.net.www.protocol.http.HttpURLConnection.getErrorStream()`